core/capabilities/ccip/ccipsolana/msghasher.go

```diff
@@ -20,12 +20,14 @@ import (
 // Compatible with:
 // - "OnRamp 1.6.0-dev"
 type MessageHasherV1 struct {
-	lggr logger.Logger
+	lggr           logger.Logger
+	extraDataCodec cciptypes.ExtraDataCodec
 }
 
-func NewMessageHasherV1(lggr logger.Logger) *MessageHasherV1 {
+func NewMessageHasherV1(lggr logger.Logger, extraDataCodec cciptypes.ExtraDataCodec) *MessageHasherV1 {
 	return &MessageHasherV1{
```

Comment:
not sure we need to pass this in, how about initializing it automatically
```suggestion
func NewMessageHasherV1(lggr logger.Logger) *MessageHasherV1 {
	extraDataCodec := ccipcommon.NewExtraDataCodec()
	return &MessageHasherV1{
```

Reply:
This does make mocking easier

Reply:
Do you need to mock the encoder? It looks like the test is doing the full borsh encoding of the extra args, so the real one might work

Reply:
If it's EVM-Solana, we would need ABI encoding, which will introduce some evm related logic in under the ccipsolana.